src/UI/Art/SpriteArtist/UnitSpriteSet.java

```diff
@@ -3,119 +3,95 @@
 import java.awt.Color;
 import java.awt.Graphics;
 import java.awt.image.BufferedImage;
-import java.awt.image.RasterFormatException;
 import java.util.ArrayList;
+import java.util.function.Function;
 
 import CommandingOfficers.Commander;
 import UI.UIUtils;
+import UI.Art.Animation.GameAnimation.AnimState;
 import Units.Unit;
 
 public class UnitSpriteSet
 {
+  Sprite sprites[] = new Sprite[AnimState.values().length];
 
-  Sprite sprites[] = new Sprite[6];
-
-  public final int ACTION_IDLE = 0;
-  public final int ACTION_MOVENORTH = 1;
-  public final int ACTION_MOVEEAST = 2;
-  public final int ACTION_MOVESOUTH = 3;
-  public final int ACTION_MOVEWEST = 4;
-  public final int ACTION_DIE = 5;
-  
-  public final int ANIM_FRAMES_PER_MARK = 3; 
-
-  Sprite turnDone;
+  public final int ANIM_FRAMES_PER_MARK = 3;
 
   /**
-   * UnitSpritSet constructor - parse the provided sprite sheet. Sprites are expected to 
-   * be arranged in rows by animation type, in the following order:
-   * idle
-   * Move North
-   * Move East
-   * Move South
-   * Move West
-   * Death
-   * 
-   * @param spriteSheet The image containing the frames to be used in this sprite set.
-   * @param width The width of each frame in the sprite.
-   * @param height The height of each frame in the sprite.
+   * Fetch a sprite sheet for each animation state and colorize it
+   * @param fileFinder Mapping from action type to file name
+   * @param coColors What colors the end sprites should use
    */
-  public UnitSpriteSet(BufferedImage spriteSheet, int width, int height, ColorPalette coColors)
+  public UnitSpriteSet(Function<AnimState, String> fileFinder, ColorPalette coColors)
```

Comment:
Why bust out a lambda function when you could just pass in a templated string?

I know we have officially moved to Java 8, but that doesn't mean we need to pull in as many shiny new toys as possible, especially when they involve distributing pointers to private member functions.

Reply:
Handing out a pointer to a private member is more private than just making the member public, which is the only thing I could do other than copy/pasting code.
...and besides, I like lambdas.

Reply:
I feel like most of the time I've seen people use lambdas it's because they think they are clever, and not because they add any value. This... doesn't seem to be an exception.
It adds complexity in order to reduce comprehensibility and increase coupling. That's a three-fer, and not in the right direction. Why not either a) partially fill out the string and pass it to `UnitSpriteSet` or b) just make `UnitSpriteSet` the owner of the string in the first place, since it's loading the actual file at this point anyway. Actually I like that better; nobody else needs to know the string or file exists at all. I may go ahead and add yet one more PR to our absurd chain here.